sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/PruneFileSourcePartitionsSuite.scala

```diff
@@ -66,4 +67,33 @@ class PruneFileSourcePartitionsSuite extends QueryTest with SQLTestUtils with Te
       }
     }
   }
+
+  test("SPARK-20986 Reset table's statistics after PruneFileSourcePartitions rule") {
+    withTempView("tempTbl", "partTbl") {
+      spark.range(1000).selectExpr("id").createOrReplaceTempView("tempTbl")
+      sql("CREATE TABLE partTbl (id INT) PARTITIONED BY (part INT) STORED AS parquet")
```

Comment:
Hi, @lianhuiwang .
`withTable("partTbl")` instead of `withTempView(..., "partTbl")`?